webknossos-datastore/app/com/scalableminds/webknossos/datastore/services/ApplicationHealthService.scala

```diff
@@ -0,0 +1,28 @@
+package com.scalableminds.webknossos.datastore.services
+
+import com.scalableminds.webknossos.datastore.slacknotification.DSSlackNotificationService
+
+import java.time.Instant
+import java.time.temporal.ChronoUnit
+import javax.inject.Inject
+
+case class HealthProblem(timestamp: Instant, msg: String)
+
+class ApplicationHealthService @Inject()(slackNotificationService: DSSlackNotificationService) {
+  private var encounteredProblems: List[HealthProblem] = List()
+
+  def pushError(e: InternalError): Unit = {
+    encounteredProblems = encounteredProblems :+ HealthProblem(Instant.now(), e.getMessage)
+    slackNotificationService.notifyForInternalError(e)
+  }
+
+  def getRecentProblem(duration: Integer = 30): Option[HealthProblem] = {
+    if (encounteredProblems.isEmpty) return None
+    if (ChronoUnit.SECONDS.between(encounteredProblems.last.timestamp, Instant.now()) <= duration) {
+      Some(encounteredProblems.last)
+    } else {
+      None
+    }
```

Comment:
Two suggestions
 - use prepend rather than append in the list (way more performant, makes headOption use possible, which is safer than last)
 - use FiniteDuration, which plays nicely with our Instant class, no need for ChronoUnit. `import scala.concurrent.duration.DurationInt` should make the `30 seconds` syntax possible.

```suggestion
  def getRecentProblem(duration: FiniteDuration = 30 seconds): Option[HealthProblem] = {
    encounteredProblems.headOption.filter { problem => 
!(problem.timestamp + duration).isPast
}
```
(did not test this code, may contain errors, but you get the idea)

Reply:
This use of filter on an option to get an option is really nice!